Bottom: fd1a09e23a94cfda0e6afe853e8a37d013669a23
Top:    6a373deee683d2682d3a8b47080e78ed6aac41aa
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-10 16:59:37 -0500

Populate buildbot with gitlab public keys


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index e14b93b9..63cebaa2 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -7,12 +7,17 @@
   become_method: sudo
   tags:
   - buildbot
+  pre_tasks:
+  - name: Scan Gitlab ssh keys
+    command: ssh-keyscan {{git_hostname}}
+    register: gitlab_keys
   roles:
   - name: Build and start the Buildbot master container
     role: buildbot-master
     tags:
     - buildbot-master
-
+    vars:
+      known_hosts: "{{gitlab_keys.stdout}}"
   - name: Add buildbot user to Gitlab
     role: gitlab-user
     vars:
diff --git a/docker/buildbot-master/buildbot/start.sh b/docker/buildbot-master/buildbot/start.sh
index 85c28248..44ec2c12 100755
--- a/docker/buildbot-master/buildbot/start.sh
+++ b/docker/buildbot-master/buildbot/start.sh
@@ -8,6 +8,12 @@ BUILDBOT_DATA=${BUILDBOT_DATA-"/buildbot"}
 
 ssh-keygen -q -b ${SSH_BITS} -t ${SSH_TYPE} -N '' -f ${SSH_FILE} 0>&- 1>/dev/null 2>/dev/null
 
+if [ -n "$BUILDBOT_KNOWN_HOSTS_FILE" ] && [ -f "$BUILDBOT_KNOWN_HOSTS_FILE" ]
+then
+    mkdir -p "${HOME}/.ssh"
+    ln -s "$BUILDBOT_KNOWN_HOSTS_FILE" "${HOME}/.ssh/known_hosts"
+fi
+
 ln -v -s -f ${HOME}/buildbot.tac ${HOME}/master.cfg ${HOME}/*.py $BUILDBOT_DATA
 
 # Run original buildbot entrypoint
